### ISSUE TITLE
M3uParser.parse does not need to be async

### ISSUE THREAD

**Issue by julemand101** [bmoliveira/m3u-parse-utils#1]
It seems like a weird design choice to make the parse method async since it does not contain any IO operations or other async stuff.

Could you remove the async or are there some reason for it that I don't see?

**Comment by timekone** [comment]
I think this would be a welcome change. But unfortunately, removing async will create a breaking change.

**Comment by julemand101** [comment]
Yeah, but I don't think it is going to be a too big of a breaking change since it can easily be fixed by removing an `await` keyword. It would be a lot worse if the opposite was going to happen where we added the need of `await`.

But I would suggest doing this in an update where other major stuff are going to be introduced or refactored.
